src/collection.js

```diff
@@ -571,22 +574,23 @@ export default class Collection {
    */
   resetSyncStatus() {
     let _count;
-    // XXX filter by status
-    return this.list({}, {includeDeleted: true})
-      .then(result => {
+    return Promise.all([
+      this.list({order: "", filters: {_status: "deleted"}}, {includeDeleted: true}),
+      this.list({order: "", filters: {_status: "synced"}}, {includeDeleted: true}),
+    ])
```

Comment:
As we're using the same query type later, we could even extract the function as something reusable:

```js
function listStatus(coll, status, options={includeDeleted: true}) {
  return coll.list({order: "", filters: {_status: status}}, options);
};
```

Or if we want to pick all records matching provided statuses:

```js
function listStatuses(coll, statuses, options={includeDeleted: true}) {
  return Promise.all(statuses.map(status => {
    return coll.list({order: "", filters: {_status: status}}, options);
  })).then((...results) => [].concat.apply([], results));
};
```